services/worker-manager/src/api.js

```diff
@@ -839,6 +841,137 @@ builder.declare({
   return res.reply({ expires: expires.toJSON(), credentials, secret: newSecret });
 });
 
+builder.declare({
+  method: 'get',
+  route: '/provisioners/:provisionerId/worker-types/:workerType/workers',
+  query: {
+    ...paginateResults.query,
+    quarantined: /^(true|false)$/,
+    requested: /^(true|false)$/,
+    running: /^(true|false)$/,
+    stopping: /^(true|false)$/,
+    stopped: /^(true|false)$/,
+  },
+  name: 'listWorkers',
+  scopes: 'worker-manager:list-workers:<provisionerId>/<workerType>',
+  stability: APIBuilder.stability.experimental,
+  category: 'Worker Metadata',
+  output: 'list-workers-response.yml',
+  title: 'Get a list of all active workers of a workerType',
+  description: [
+    'Get a list of all active workers of a workerType.',
+    '',
+    '`listWorkers` allows a response to be filtered by quarantined and non quarantined workers,',
+    'as well as the current state of the worker.',
+    'To filter the query, you should call the end-point with one of [`quarantined`, `requested`, `running`,',
+    '`stopping`, `stopped`] as a query-string option with a true or false value.',
+    '',
+    'The response is paged. If this end-point returns a `continuationToken`, you',
+    'should call the end-point again with the `continuationToken` as a query-string',
+    'option. By default this end-point will list up to 1000 workers in a single',
+    'page. You may limit this with the query-string parameter `limit`.',
+  ].join('\n'),
+}, async function(req, res) {
+  const quarantined = req.query.quarantined || null;
+  const requested = req.query.requested || null;
+  const running = req.query.running || null;
+  const stopping = req.query.stopping || null;
+  const stopped = req.query.stopped || null;
+  const workerState = requested || running || stopping || stopped || null;
+  const { provisionerId, workerType } = req.params;
+  const now = new Date();
+  const taskQueueId = joinTaskQueueId(provisionerId, workerType);
```

Comment:
So once you will switch to worker's manager implementation it would  be named `workerPoolId`. Because `taskQueueId` is terminology from the Queue service ;)

Check [docs](https://firefox-ci-tc.services.mozilla.com/docs/reference/platform/queue/worker-hierarchy#worker-pools-and-task-queues)